src/vm/codeman.cpp

```diff
@@ -1302,121 +1303,128 @@ void EEJitManager::SetCpuInfo()
-        // CORJIT_FLAG_USE_SSE3 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        // CORJIT_FLAG_USE_SSSE3 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        // CORJIT_FLAG_USE_SSE41 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.1 - ECX bit 19  (buffer[10] & 0x08)
-        // CORJIT_FLAG_USE_SSE42 if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.2 - ECX bit 20  (buffer[10] & 0x10)
-        // CORJIT_FLAG_USE_POPCNT if the following feature bits are set (input EAX of 1)
-        //    SSE3 - ECX bit 0     (buffer[8]  & 0x01)
-        //    SSSE3 - ECX bit 9    (buffer[9]  & 0x02)
-        //    SSE4.2 - ECX bit 20  (buffer[10] & 0x10)
-        //    POPCNT - ECX bit 23  (buffer[10] & 0x80)
-        // CORJIT_FLAG_USE_AVX if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    OSXSAVE - ECX bit 27 (buffer[11] & 0x08)
-        //    AVX - ECX bit 28     (buffer[11] & 0x10)
-        // CORJIT_FLAG_USE_FMA if the following feature bits are set (input EAX of 1), and xmmYmmStateSupport returns 1:
-        //    FMA - ECX bit 12     (buffer[9]  & 0x10)
-        // CORJIT_FLAG_USE_AVX2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    AVX2 - EBX bit 5     (buffer[4]  & 0x20)
-        // CORJIT_FLAG_USE_AVX_512 is not currently set, but defined so that it can be used in future without
-        // CORJIT_FLAG_USE_BMI1 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    BMI1 - EBX bit 3     (buffer[4]  & 0x08)
-        // CORJIT_FLAG_USE_BMI2 if the following feature bit is set (input EAX of 0x07 and input ECX of 0):
-        //    BMI2 - EBX bit 8     (buffer[5]  & 0x01)
-        // CORJIT_FLAG_USE_LZCNT if the following feature bits are set (input EAX of 80000001H)
-        //    LZCNT - ECX bit 5    (buffer[8]  & 0x20)
-        // synchronously updating VM and JIT.
+
         (void) getcpuid(1, buffer);
-        // If SSE2 is not enabled, there is no point in checking the rest.
-        // SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
+
+        // If SSE/SSE2 is not enabled, there is no point in checking the rest.
+        //   SSE  is bit 25 of EDX   (buffer[15] & 0x02)
+        //   SSE2 is bit 26 of EDX   (buffer[15] & 0x04)
         // TODO: Determine whether we should break out the various SSE options further.
-        if ((buffer[15] & 0x04) != 0)               // SSE2
+        if ((buffer[15] & 0x06) == 0x06)                                    // SSE & SSE2
```

Comment:
We don't seem to have any handling for the case when `SSE/SSE2` are not set.

Reply:
Interesting point. I believe it was the case for the desktop implementation that it could be assumed, since it was (IIRC) a minimum requirement for Windows. However, for coreclr perhaps there should be some checking.

Reply:
Right. For (all supported) buildsof Windows: SSE/SSE2 (and some additional things, like CMPXCHG16B, PREFETCHW, and LAHF/SAHF) are required.

I don't think that requirement is hardset for all Linux distro's however, so some minimal handling to say "we require SSE/SSE2" would probably be good.